lib/packagekit-glib2/pk-offline.c

```diff
@@ -722,18 +725,19 @@ pk_offline_get_results (GError **error)
 		pk_results_set_role (results, pk_role_enum_from_string (role_str));
 
 	/* add packages */
-	data = g_key_file_get_string (file, PK_OFFLINE_RESULTS_GROUP,
-				      "Packages", NULL);
-	if (data != NULL) {
-		package_ids = g_strsplit (data, ",", -1);
-		for (i = 0; package_ids[i] != NULL; i++) {
-			g_autoptr(PkPackage) pkg = NULL;
-			pkg = pk_package_new ();
-			pk_package_set_info (pkg, PK_INFO_ENUM_UPDATING);
-			if (!pk_package_set_id (pkg, package_ids[i], error))
-				return NULL;
-			pk_results_add_package (results, pkg);
-		}
+	package_ids = g_key_file_get_string_list (file, PK_OFFLINE_RESULTS_GROUP,
+						  "Packages", &package_ids_size, NULL);
+	if (package_ids == NULL || package_ids_size == 0)
+		goto end;
+
+	for (i = 0; package_ids[i] != NULL; i++) {
+		g_autoptr(PkPackage) pkg = NULL;
+		pkg = pk_package_new ();
+		pk_package_set_info (pkg, PK_INFO_ENUM_UPDATING);
+		if (!pk_package_set_id (pkg, package_ids[i], error))
+			return NULL;
+		pk_results_add_package (results, pkg);
 	}
+end:
```

Comment:
nit: We commonly call this `out` or `finish` in other parts of the code - definitely a nitpick though.